cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AbstractCompositeTrigger.java

```diff
@@ -85,4 +88,14 @@ private void initializeUnitTriggers() {
       }
     }
   }
+
+  public List<TriggerInfo> getUnitTriggerInfos(TriggerInfoContext context) {
```

Comment:
Or, if the field is not final, then the Map constructed Also, comment on line 36 is wrong, as a composite trigger can have another composite trigger inside, right?

Also, if you use `LinkedListMultimap` to store the triggers, then you can simply return an unmodifiable list of the multimap.values(), without the need of copying.